src/wp-includes/class-wp-metadata-lazyloader.php

```diff
@@ -154,16 +149,44 @@ public function lazyload_term_meta( $check ) {
 	 * directly, from either inside or outside the `WP_Query` object.
 	 *
 	 * @since 4.5.0
+	 * @deprecated 6.3.0
 	 *
 	 * @param mixed $check The `$check` param passed from the {@see 'get_comment_metadata'} hook.
 	 * @return mixed The original value of `$check`, so as not to short-circuit `get_comment_metadata()`.
 	 */
 	public function lazyload_comment_meta( $check ) {
-		if ( ! empty( $this->pending_objects['comment'] ) ) {
-			update_meta_cache( 'comment', array_keys( $this->pending_objects['comment'] ) );
+		_deprecated_function( __METHOD__, '6.3.0', 'WP_Metadata_Lazyloader::lazyload_meta_callback' );
+		return $this->lazyload_meta_callback( $check, 0, '', false, 'comment' );
+	}
+
+	/**
+	 * Lazy-loads meta for queued objects.
+	 *
+	 * This method is public so that it can be used as a filter callback. As a rule, there
+	 * is no need to invoke it directly.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param mixed  $check     The `$check` param passed from the 'get_*_metadata' hook.
+	 * @param int    $object_id ID of the object metadata is for.
+	 * @param string $meta_key  Unused.
+	 * @param bool   $single    Unused.
+	 * @param string $meta_type Type of object metadata is for. Accepts 'post', 'comment', 'term', 'user',
+	 *                          or any other object type with an associated meta table.
+	 * @return mixed In order not to short-circuit `get_metadata()`. Generally, this is `null`, but it could be
+	 *               another value if filtered by a plugin.
+	 */
+	public function lazyload_meta_callback( $check, $object_id, $meta_key, $single, $meta_type ) {
+		if ( ! empty( $this->pending_objects[ $meta_type ] ) ) {
```

Comment:
Nitty nit non-blocking nit...

To avoid all excessive indentation, I usually do an early return in situations like this:

```php
if ( empty( $this->pending_objects[ $meta_type ] ) ) {
  return $check;
}

// Do the lazy loading

return $check;
```

Thoughts?